cmd/forwarder/http.go

```diff
@@ -147,6 +150,18 @@ func (s *httpServer) Run() error {
 			defer body.Close()
 		}
 
+		buf := new(bytes.Buffer)
+		// This should only be reached if authentication information is valid.
+		authUser, _, _ := r.BasicAuth()
+		if s.Config.LogAuthUser() &&
+			authUser != "" &&
+			authUser != s.Config.ValidTokenUser &&
+			rand.Intn(99) > s.Config.TokenUserSamplePct {
+
+			// tee body to buffeer
+			body = ioutil.NopCloser(io.TeeReader(body, buf))
```

Comment:
The change above to the creation of buf will require you to use `&buf` (take the pointer of buf) here.